backend/src/main/java/quarano/security/web/AuthenticationController.java

```diff
@@ -60,17 +68,28 @@
 						.ifPresent(people::save))
 				.<HttpEntity<?>> map(accountRepresentations::toTokenResponse)
 				.recover(EmptyResultDataAccessException.class, it -> toUnauthorized(it.getMessage()))
+				.recover(ExternalZipException.class, it -> toUnprocessable(it.getMessage()))
 				.get();
 	}
 
 	private boolean hasOpenCaseForAccount(Account account) {
 
-		return !account.isTrackedPerson() || people.findByAccount(account)
-				.flatMap(cases::findByTrackedPerson)
+		return !account.isTrackedPerson() || loadCaseFor(account)
 				.filter(TrackedCase::isOpen)
 				.isPresent();
 	}
 
+	private boolean isntCaseExternal(Account account) {
+
+		return !account.isTrackedPerson() || loadCaseFor(account)
+				.filter(it -> it.getStatus() != Status.EXTERNAL_ZIP)
+				.isPresent();
+	}
+
+	private Optional<TrackedCase> loadCaseFor(Account account) {
+		return people.findByAccount(account).flatMap(cases::findByTrackedPerson);
```

Comment:
Does it make sense to introduce a dedicated query on `TrackedCaseRepository` that looks up the case for an account directly? That would avoid two query executions one after the other in favor of just one. Also, does it make sense to reuse the `TrackedCase` instance obtained in the `hasOpenCaseForAccount(…)` check? We seem to be looking up the case *again* for the second check when it's in fact the same case we need to check.

Reply:
First question: :+1: 
Second question: I search a solution.

Reply:
We might have to move away from the nice constraint per line style and rather have a single, slightly more complex, single `filter(…)` expression that reuses the initially obtained instance.

Reply:
I think we have two options:

1.
```
.map(it -> Tuple.of(it, cases.findByAccount(it)))
.filter(this::hasOpenCaseForAccount,
() -> new AccessDeniedException(message.getMessage("authentication.trackedCase.closed")))
this::isntCaseExternal,
() -> new ExternalZipException(message.getMessage("authentication.trackedCase.externalZip")))
.map(it -> it._1)
```

2. We must use one `Predicate` and then thrown the special Exception inside this.

I prefer the first option.

Reply:
is revised